node/core/pvf/common/src/worker/mod.rs

```diff
@@ -317,4 +317,85 @@ pub mod thread {
 			Some(*result.0)
 		}
 	}
+
+	#[cfg(test)]
+	mod tests {
+		use super::*;
+
+		#[test]
+		fn get_condvar_should_be_pending() {
+			let condvar = get_condvar();
+			let outcome = *condvar.0.lock().unwrap();
+			assert!(outcome.is_pending());
+		}
+
+		#[test]
+		fn wait_for_threads_with_timeout_return_none_on_time_out() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let outcome = wait_for_threads_with_timeout(&condvar, Duration::from_secs(1));
+			assert!(matches!(outcome, None));
+		}
+
+		#[test]
+		fn wait_for_threads_with_timeout_returns_outcome() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let condvar2 = Arc::clone(&condvar);
+			thread::spawn(move || {
+				cond_notify_all(Arc::clone(&condvar2), WaitOutcome::Finished);
+			});
+			let outcome = wait_for_threads_with_timeout(&condvar, Duration::from_secs(2));
+			assert!(matches!(outcome.unwrap(), WaitOutcome::Finished));
+		}
+
+		#[test]
+		fn spawn_worker_thread_should_notify_on_done() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let response =
+				spawn_worker_thread("thread", || 2, Arc::clone(&condvar), WaitOutcome::TimedOut);
+			let (lock, _) = &*condvar;
+			let r = response.unwrap().join().unwrap();
+			assert_eq!(r, 2);
+			assert!(matches!(*lock.lock().unwrap(), WaitOutcome::TimedOut));
+		}
+
+		#[test]
+		fn spawn_worker_thread_should_not_notify() {
```

Comment:
test name is a little bit misleading, you don't check for condvar not being notified, but instead that the flag remains unchanged since it wasn't `Pending`.
Perhaps `worker_only_changes_pending_outcome` or smth